src/cr/cube/matrix/cubemeasure.py

```diff
@@ -268,6 +298,42 @@ def table_bases(self):
             "`%s` must implement `.table_bases`" % type(self).__name__
         )
 
+    @lazyproperty
+    def _columns_pruning_base(self):
+        """1D bool np.ndarray of the sum of the column-bases for cells in a column
+
+        Used to calculate he columns-pruning-mask, which indicates when every cell in a
+        column has base values of 0. When the column is *not* an MR, the base used is
+        equal to the columns_base, but when the column is an MR, we use the sum of
+        selected & non-selected values the column, whereas the columns_base uses just the
+        selected values. These values are not meaningful on their own, but used to
+        calculate the columns-pruning-mask.
+
+        This method works when the column is not an MR, and must be overriden when it is
+        MR.
+        """
+        # --- Bases are positive, so we can sum them to see if all of them are equal to 0
+        # --- To really hammer home the point that this number is invalid, we keep it
+        # --- private. But it's easier to test, so it is separate from the mask.
+        return np.sum(self.column_bases, axis=0)
+
+    @lazyproperty
+    def _rows_pruning_base(self):
+        """1D bool np.ndarray of the sum of the row-bases for cells in a row
+
+        Used to calculate he rows-pruning-mask, which indicates when every cell in a row
+        of 0. When the row is *not* an MR, the base used is equal to the rows_base,
+        but when the row is an MR, we use the sum of selected & non-selected values from
+        the row, whereas the rows_base uses just the selected values. These values
+        are not meaningful on their own, but used to calculate the rows-pruning-mask.
+
+        This method works when the row is not an MR, and must be overriden when it is MR.
+        """
+        # --- Bases are positive, so we can sum them to see if all of them are equal to 0
+        # --- To really hammer home the point that this number is invalid, we keep it
+        # --- private. But it's easier to test, so it is separate from the mask.
```

Comment:
Okay, I never thought I'd say this but I think we need to work out a way to _reduce_ the documentation here. There's so much redundancy that you have to dig to find out what's different.

I think a decent approach is to be thorough in the base class (which might not even have an implementation but at least is "centralized"), and then concise in the individual subclass implementations. I'd say a concise form is from this:

```
    @lazyproperty
    def _rows_pruning_base(self):
        """1D bool np.ndarray of the sum of the row-bases for cells in a row

        Used to calculate he rows-pruning-mask, which indicates when every cell in a row
        of 0. When the row is *not* an MR, the base used is equal to the rows_base,
        but when the row is an MR, we use the sum of selected & non-selected values from
        the row, whereas the rows_base uses just the selected values. These values
        are not meaningful on their own, but used to calculate the rows-pruning-mask.

        This method works when the row is not an MR, and must be overriden when it is MR.
        """
        # --- Bases are positive, so we can sum them to see if all of them are equal to 0
        # --- To really hammer home the point that this number is invalid, we keep it
        # --- private. But it's easier to test, so it is separate from the mask.
        return np.sum(self.row_bases, axis=1)
```

to this:

```
    @lazyproperty
    def _rows_pruning_base(self):
        """1D bool np.ndarray of the sum of the row-bases for cells in a row.

        Used to compute the rows-pruning-mask; not meaningful on its own.
        """
        return np.sum(self.row_bases, axis=1)
```

I naturally consult the base-class documentation as a second source, but if we think others won't we could add "see base class for more details" or whatever.

Reply:
Ha, there's a Berenstain bears book where the kids try to be so polite that it'll drive the Mom crazy, but it backfires and they end up forgetting about the plan and just become "regular" polite. Hopefully that'll happen to me. (Makes sense about having the base class as the central place, as I wrote that, I wasn't sure how we felt about doing that).